src/main/resources/xsl/openagrar-solr.xsl

```diff
@@ -60,19 +59,41 @@
         </field>
       </xsl:if>
     </xsl:for-each>
-    <xsl:for-each
+
+	<xsl:for-each
       select="mods:name[mods:role/mods:roleTerm[@authority='marcrelator' and (@type='text' and text()='author') or (@type='code' and text()='aut')]]">
-      <xsl:if test="position()= last()">
-        <field name="mods.lastAuthor">
-          <xsl:for-each select="mods:displayForm | mods:namePart | text()">
-            <xsl:value-of select="concat(' ',mcrxsl:normalizeUnicode(.))" />
-          </xsl:for-each>
-        </field>
-        <field name="mods.lastAuthor.affiliation">
-          <xsl:value-of select="mods:affiliation" />
-        </field>
-      </xsl:if>
-    </xsl:for-each>
+		  <xsl:variable name="autrole">
+			  <xsl:choose>
+				<xsl:when test="./mods:role/mods:roleTerm[substring-after(@valueURI, '#')='main_author']">
+					<xsl:text>mainAuthor</xsl:text>
+				</xsl:when>
+				<xsl:when test="count(//mods:roleTerm[substring-after(@valueURI, '#')='main_author'])=0 and position()=1">
```

Comment:
Bitte die Bedingung
//mods:roleTerm[substring-after(@valueURI, '#')='main_author'])=0
nach 
//mods:roleTerm[substring-after(@valueURI, '#'))=0
ändern.

Damit wird verhindert, dass beide Varianten gleichzeitig genutzt werden.

Reply:
Verstehe ich nicht.  Wieso würden mit diesem Code beide Varianten gleichzeitig benutzt?

Reply:
### Bsp. 1 Es gibt nur einen MainAuthor
**dokument**
- Person 1 (#main_author) 

**solrdocument (derzeit)** 
- mods.mainAuthor=Person 1
- mods.lastAuthor=Person 1 

### Bsp. 2 1 MainAuthor 2 CoAutoren
**dokument**
- Person 1 (#main_author) 
- Person 2 (#co_author) 
- Person 3 (#co_author) 

**solrdocument (derzeit)** 
- mods.mainAuthor=Person 1
- mods.coAuthor=Person 2
- mods.coAuthor=Person 3
- mods.lastAuthor=Person 3

Reply:
Es muss nicht immer einen last oder einen main Autor geben.

Reply:
Außerdem keine globalen xpaths mit //, da mehrere dokumente auf einmal indexiert werden. Wenn dann .// bzw. relative pfade.

Reply:
Ja, Bsp.1 und 2 habe ich meines Wissens berücksichtigt. Ich prüfe zuerst, ob "main_author" vorhanden ist, falls ja, wird die Person als solche ausgezeichnet. Falls nein, prüfe ich ob überhaupt eine Person "main_autor" ist, falls nicht wird die Person als "main_autor" ausgezeichenet, sofern sie die erste in der Personenreihe ist, sonst werden weitere Prüfungen gemacht.

Reply:
Dass globale xpaths nicht verwendet werden soll, wusste ich nicht. Dann muss ich es anders lösen.